bin/grommet.js

```diff
@@ -17,76 +32,251 @@ String.prototype.capitalize = function() {
+    return function() {
+      if (!dependenciesSupported(command.env.nodeVersion, command.env.npmVersion)) {
+        process.exit(1);
+      }
+
+      var newAppPath = path.join(process.cwd(), command.options.app);
+
+      mkdirp(newAppPath, function(err) {
+        if (err) {
+          console.log('Error trying to create project: ' + err);
+        } else {
+          var templateFolder = path.join(command.env.templatePath, 'init/**');
+          var mobileIcon = path.join(command.env.grommetPath, 'img/mobile-app-icon.png');
+          var shortcutIcon = path.join(command.env.grommetPath, 'img/shortcut-icon.png');
+
+          gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
+          gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        }
+      })
     }
+  },
+  'new': function(command) {
+    return function() {
+      if (!dependenciesSupported(command.env.nodeVersion, command.env.npmVersion)) {
+        process.exit(1);
+      }
 
-    mkdirp('./' + app, function(err) {
-      if (err) {
-        console.log('Error trying to create project: ' + err);
-      } else {
-        process.chdir('./' + app);
+      var newAppPath = path.join(process.cwd(), command.options.app);
 
-        var templateFolder = path.join(grommetPath, 'templates/init/**');
-        var mobileIcon = path.join(grommetPath, 'img/mobile-app-icon.png');
-        var shortcutIcon = path.join(grommetPath, 'img/shortcut-icon.png');
+      if (fileExists(newAppPath)) {
+        console.error('[grommet] Error while creating new app. Directory "'+command.options.app+'" already exists.');
+        process.exit(1);
+      }
 
-        gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(templateFolder, { dot: true })
-        .pipe(template({
-          appName: app,
-          appTitle: title,
-          grommetVersion: packageJSON.version
-        }))
-        .pipe(gulp.dest('./'))
-        .pipe(install()).on('finish', function() {
-          done();
+      mkdirp(newAppPath, function(err) {
+        if (err) {
+          console.error('[grommet] Error trying to create project: ' + err);
+          process.exit(1);
+        }
+
+        try {
+          var templateFolder = path.join(command.env.templatePath, 'new/**');
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        } catch(err) {
+          shelljs.rm('-rf', newAppPath);
+          console.error('[grommet] Error while creating new app '+command.options.app+'.');
+          throw err;
+        }
+      })
+    }
+  },
+  'create-component': function(command) {
+    return function() {
+      var methodBlacklist = [ 'constructor', 'render' ];
+      var componentTemplate = 'component_template.js';
+      var from = path.join(command.env.templatePath, componentTemplate);
+      var to = path.join(process.cwd(), 'src/js/components');
```

Comment:
i would assume that we would create the component in the current folder. we cannot rely that `src/js/components` will be always present.

So, what do you think about `var to = process.cwd();`